command/push.py

```diff
@@ -1,8 +1,35 @@
+import base64
+import os
+import requests
+import tarfile
+
 class PushCmd():
     name = 'push'
 
     def __init__(self):
         pass
         
-    def push(self):
-        print("Hello, world. The push feature is currently under construction.")
+    def push(self, bundle_dir, namespace, repository, release, auth_token):
+        base64_bundle = self.__create_base64_bundle(bundle_dir, repository)
+        self.__push_to_registry(namespace, repository, release, base64_bundle, auth_token)
+
+    def __create_base64_bundle(self, bundle_dir, repository):
+        tarfile_name = "%s.tar.gz" % repository
+        with tarfile.open(tarfile_name, "w:gz") as tar:
+            tar.add(bundle_dir, arcname=os.path.basename(bundle_dir))
+        with open(tarfile_name, "rb") as tarball:
+            result = tarball.read()
+        result64 = base64.b64encode(result).decode("utf-8")
+        return result64
+
+    def __push_to_registry(self, namespace, repository, release, bundle, auth_token):
+        push_uri = 'https://quay.io/cnr/api/v1/packages/%s/%s' % (namespace, repository)
+        print("Pushing bundle to %s" % push_uri)
+        headers = {'Content-Type': 'application/json', 'Authorization': auth_token}
+        json = {'blob': bundle, 'release': release, "media_type": "helm"}
+        r = requests.post(push_uri, json=json, headers=headers)
+        if r.status_code == 200:
+            print("Pushed bundle to registry.")
+        else:
+            print("Failed to push to app registry.")
```

Comment:
This should raise an exception, not print to stdout.

Reply:
Good point. Updated the pr to raise an exception there.